mlir/test/Dialect/Vector/int-range-interface.mlir

```diff
@@ -0,0 +1,57 @@
+// RUN: mlir-opt -int-range-optimizations -canonicalize %s | FileCheck %s
+
+
+// CHECK-LABEL: func @constant_vec
+// CHECK: test.reflect_bounds {smax = 7 : index, smin = 0 : index, umax = 7 : index, umin = 0 : index}
+func.func @constant_vec() -> vector<8xindex> {
+  %0 = arith.constant dense<[0, 1, 2, 3, 4, 5, 6, 7]> : vector<8xindex>
+  %1 = test.reflect_bounds %0 : vector<8xindex>
```

Comment:
I don't really know what these test Ops do and I couldn't find any documentation in code. Could add some docs, pls?

Reply:
These test ops are from existing integer range inference tests - they have an implement of the integer range inference interface that sets attributes on `reflect_bounds` to match the bounds of the input

Reply:
Yes, I gathered that much from mlir/test/lib/Dialect/Test/TestOps.td, but it doesn’t quite clarify things for me. The lack of documentation for these operations makes it hard to understand the distinction between `test.reflect_bounds` and `test.with_bounds`. 

Given that @Hardcode84 is already using these ops for testing, it would be fantastic if some of that expertise could be shared through documentation. This would benefit everyone working with these tests!

Reply:
added some description to the ops